### PR TITLE
[go-gen] Add delete handler generation

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of delete handlers
  * Should handle readable/writable/auth/project-auth cases
* Improves checkAuthorization generation
  * Removes it from being generated for services with an auth block, as the ID is a one-to-one mapping with AuthID, so the incoming AuthID can be directly compared with the resource ID to check authorization

Standard stuff

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala [hunk 1]
```diff
@@ -117,21 +117,28 @@ object GoServiceMainHandlersGenerator {
...
+          genFunctionCall("checkAuthorization", "env", s"${root.decapitalizedName}ID", "auth"),
+          "authorized",
+          "err",
         ),
-      ),
-      genIf("!authorized", generateHTTPErrorReturn(StatusUnauthorized, "Unauthorized")),
-    )
+        genIfErr(
+          genSwitchReturn(
+            "err.(type)",
```
Comment (c-harding) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala:134]: wat

**Review by c-harding — APPROVED** [review]
LGTM

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala:134]: nice syntax right

**Review by jaylees14 — APPROVED** [review]

**Pull request merged**